### PR TITLE
refactor help func in CLI

### PR DESCRIPTION
**Pull request by vieux**
ping @dnephin

### CODE REVIEW

File: cli/command/container/cmd.go [hunk 1]
```diff
@@ -13,10 +13,7 @@ func NewContainerCommand(dockerCli *command.DockerCli) *cobra.Command {
 		Use:   "container",
 		Short: "Manage containers",
 		Args:  cli.NoArgs,
-		Run: func(cmd *cobra.Command, args []string) {
-			cmd.SetOutput(dockerCli.Err())
-			cmd.HelpFunc()(cmd, args)
-		},
+		RunE:  dockerCli.ShowHelp,
```
Comment (AkihiroSuda) [inline cli/command/container/cmd.go]: Why RunE

File: cli/command/checkpoint/cmd.go [hunk 2]
```diff
@@ -12,11 +12,8 @@ func NewCheckpointCommand(dockerCli *command.DockerCli) *cobra.Command {
 		Use:   "checkpoint",
 		Short: "Manage checkpoints",
 		Args:  cli.NoArgs,
-		Run: func(cmd *cobra.Command, args []string) {
-			cmd.SetOutput(dockerCli.Err())
-			cmd.HelpFunc()(cmd, args)
-		},
-		Tags: map[string]string{"experimental": "", "version": "1.25"},
+		RunE:  dockerCli.ShowHelp,
```
Comment (AkihiroSuda) [inline cli/command/checkpoint/cmd.go]: Why RunE?

**Review by AkihiroSuda — APPROVED** [review]
LGTM, but nit comment above

File: cli/command/checkpoint/cmd.go (hunk 2 above)
Comment (vieux) [inline cli/command/checkpoint/cmd.go]: so every command now use `RunE` , I have a followup PR that will override cmd in some cases. it harder to do if we have a mix of `Run` and `RunE`

**Review by dnephin — APPROVED** [review]
LGTM

**Comment by tonistiigi** [comment]
Test failures unrelated.

File: cli/command/checkpoint/cmd.go (hunk 2 above)
Comment (AkihiroSuda) [inline cli/command/checkpoint/cmd.go]: SGTM :+1:

**Pull request merged**